Mathlib/Init/Data/Ordering/Basic.lean

```diff
@@ -14,18 +16,28 @@ deriving instance Repr for Ordering
 
 namespace Ordering
 
-/-- Combine two `Ordering`s lexicographically. -/
-@[inline]
-def orElse : Ordering → Ordering → Ordering
-  | lt, _ => lt
-  | eq, o => o
-  | gt, _ => gt
-
-/-- The relation corresponding to each `Ordering` constructor (e.g. `.lt.toProp a b` is `a < b`). -/
-def toRel [LT α] : Ordering → α → α → Prop
-  | .lt => (· < ·)
-  | .eq => Eq
-  | .gt => (· > ·)
+@[deprecated] alias orElse := «then»
+#align ordering.or_else Ordering.then
+
+#align ordering.swap_swap Ordering.swap_swap
+
+/-- `Compares o a b` means that `a` and `b` have the ordering relation `o` between them, assuming
+that the relation `a < b` is defined. -/
+-- Porting note: we have removed `@[simp]` here in favour of separate simp lemmas,
+-- otherwise this definition will unfold to a match.
+def Compares [LT α] : Ordering → α → α → Prop
+  | lt, a, b => a < b
+  | eq, a, b => a = b
+  | gt, a, b => a > b
+#align ordering.compares Ordering.Compares
+
+@[deprecated] alias toRel := Compares
+
+@[simp] lemma compares_lt [LT α] (a b : α) : Compares lt a b = (a < b) := rfl
+
+@[simp] lemma compares_eq [LT α] (a b : α) : Compares eq a b = (a = b) := rfl
+
+@[simp] lemma compares_gt [LT α] (a b : α) : Compares gt a b = (a > b) := rfl
```

Comment:
Do we actually want to be moving things *into* `Init`? My impression was that `Init` was basically just "cruft from core that we haven't decided what to do with yet", for which the natural progression is to gradually either promote or delete declarations.

Reply:
I did this only from a avoiding-unnecessary-imports perspective. `Compares` was in a file with lots of imports and there are other `Init` files that import this and use `toRel`, now `Compares`.